org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/operation/OperationHistoryObservableValue.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Patrick Ziegler and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.pde.ui.target.editor.internal.operation;
+
+import org.eclipse.core.commands.operations.DefaultOperationHistory;
+import org.eclipse.core.commands.operations.IOperationHistory;
+import org.eclipse.core.databinding.observable.ChangeEvent;
+import org.eclipse.core.databinding.observable.value.AbstractObservableValue;
+
+/**
+ * There is currently only a single implementation for IOperationHistory, the
+ * {@link DefaultOperationHistory}. Sadly, this class stores its operations in a
+ * normal list, rather than observable lists. As a result, performing any
+ * operations on the history doesn't trigger any side-effects.<br>
+ * We work around this limitation by firing a change event whenever the history
+ * is accessed (as opposed to writing a proper "observable" history).<br>
+ * As a result, the Undo/Redo buttons are never updated. :(
+ * 
+ * @param <T> The explicit implementation of {@link IOperationHistory}.
+ */
+public class OperationHistoryObservableValue<T extends IOperationHistory> extends AbstractObservableValue<T> {
+	private final T operationHistory;
+
+	public OperationHistoryObservableValue(T operationHistory) {
+		this.operationHistory = operationHistory;
+	}
+
+	@Override
+	public Object getValueType() {
+		return null;
+	}
+
+	@Override
+	protected T doGetValue() {
+		fireEvent(new ChangeEvent(this));
```

Comment:
this seems not correct here why is getting a value firing a change event?

Reply:
That's what I meant... It's a quick & dirty solution to make sure the buttons are updated, whenever something is done to the IOperationHistory instance.

The Undo button, for example, checks its enablement state by calling IOperationHistory.canUndo(IUndoContext). 
But because this call doesn't access any IObservables, the side-effect for the button is never executed again. My "fix" was to create a custom dummy IObservableValue wrapping the IOperationHistory.  which notifies this side-effect, whenever accessed.

Reply:
You should not call the `IOperationHistory` directly but for example `OperationHistoryFacade#canUndo` (like outlined below) that then delegates to the `IOperationHistory`